### PR TITLE
Fix #2999: Add onMore to Select

### LINKED ISSUE #2999: Need a help to implement Select options component to load pagination api data
How load a pagination data in Select options? 

### Your Environment

- Grommet version: 2.4.0
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by shivamsingha**
#### What does this PR do?
Fixes #2999. Adds onMore to Select component and a storybook example for it.

#### Where should the reviewer start?
src/js/components/Select/SelectContainer.js

#### What testing has been done on this PR?
storybook

#### How should this be manually tested?
storybook

#### What are the relevant issues?
#2999 

#### Do the grommet docs need to be updated?
Yes

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/Select/SelectContainer.js [hunk 1]
```diff
@@ -390,7 +391,12 @@ class SelectContainer extends Component {
             overflow="auto"
           >
             {options.length > 0 ? (
-              <InfiniteScroll items={options} step={theme.select.step} replace>
+              <InfiniteScroll
+                items={options}
+                step={theme.select.step}
+                onMore={onMore}
+                replace
```
Comment (oorestisime) [inline src/js/components/Select/SelectContainer.js]: maybe we should add replace to props as well?

Comment (shivamsingha) [reply src/js/components/Select/SelectContainer.js]: Would this work? I'm not sure how to deal with booleans.
`<InfiniteScroll ....................
      {this.prop.replace} >`

Comment (oorestisime) [reply src/js/components/Select/SelectContainer.js]: just like you did with onMore. `replace={replace}`. and destructure it before. you might want to set  the defaultProps as well to true.

**Comment by oorestisime** [comment]
Do you mind changing the documentation as well in your PR? 
You will need to add the onMore and replace property in the doc.js file https://github.com/grommet/grommet/blob/master/src/js/components/Select/doc.js

**Comment by shivamsingha** [comment]
Am I supposed to push the snapshots too? The CI test wasn't passing without it

**Comment by oorestisime** [comment]
Yeah snapshots are stored in :) will take a look again later today! thanks again

**Comment by oorestisime** [comment]
Looking good to me! @ericsoderberghp mind taking a peek as well?

File: src/js/components/Select/stories/LazyLoading.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      const { options } = this.state;
+      console.log('onmore called');
+      this.setState({ options: dummyOptions.slice(0, options.length + 200) });
+    }, 1000);
+  };
+
+  onClose = () => {
+    const { selected, options } = this.state;
+    this.setState({
+      options: options.sort((p1, p2) => {
```
Comment (ericsoderberghp) [inline src/js/components/Select/stories/LazyLoading.js]: Why do we need to sort the options when the Select drop down is closed?

**Comment by ericsoderberghp** [comment]
This is looking great. Thanks for the work on it. Can you also add the typescript typings in Select/index.d.ts?

File: src/js/components/Select/stories/LazyLoading.js (hunk 2 above)
Comment (shivamsingha) [reply src/js/components/Select/stories/LazyLoading.js]: Maybe because of replace?
It was in manyoptions.js. I wasn't sure so didn't remove.

Comment (shivamsingha) [reply src/js/components/Select/stories/LazyLoading.js]: I think it's for moving selected options to the top.
But its not working in both this one and manyoptions.js

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Pull request merged**